### ISSUE TITLE
Scaled Cosine Attention implementation

### ISSUE THREAD

**Issue by 0reza** [microsoft/Swin-Transformer#239]
### Can someone explain me what happened here? despite what is on the paper.
The relative position bias is okay. Just don't know where is the cosine function?

https://github.com/microsoft/Swin-Transformer/blob/b720b4191588c19222ccf129860e905fb02373a7/models/swin_transformer_v2.py#L154-L157
---
[Swin Transformer V2: Scaling Up Capacity and Resolution.](https://doi.org/10.48550/arXiv.2111.09883)
![image](https://user-images.githubusercontent.com/67942779/179022917-167e0125-588a-4d8c-b828-ad84e79efcff.png)

**Comment by rami0205** [comment]
in line 155
`(F.normalize(q, dim=-1) @ F.normalize(k, dim=-1).transpose(-2, -1)) `
this code means calculating cosine similarity of Q and K
it is because the definition of cosine similarity is
cos-sim(a, b) = 
![image](https://user-images.githubusercontent.com/69415453/186560523-ca8b43f2-1e83-4016-b344-50435163ec0f.png)

`F.normalize(q, dim=-1)` means the first term and `F.normalize(k, dim=-1)` means the second term.
F.normalize function computes l2 normalization.
